src/Uno.UI.RuntimeTests/Tests/Windows_Storage/Given_StorageFolder_Native.base.cs

```diff
@@ -0,0 +1,1249 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.Storage;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_Storage
+{
+	public abstract class Given_StorageFolder_Native_Base
+	{
+		protected abstract Task<StorageFolder> GetRootFolderAsync();
+
+		protected virtual Task CleanupRootFolderAsync() => Task.CompletedTask;
+
+		[TestMethod]
+		public async Task When_CreateFolder_Name_Matches()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var folderName = Guid.NewGuid().ToString();
+			StorageFolder? createdFolder = null;
+			try
+			{
+				createdFolder = await rootFolder.CreateFolderAsync(folderName);
+				Assert.AreEqual(folderName, createdFolder.Name);
+			}
+			finally
+			{
+				if (createdFolder != null)
+				{
+					await createdFolder.DeleteAsync();
+				}
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+		public async Task When_CreateFolder_DisplayName_Matches()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var folderName = Guid.NewGuid().ToString();
+			StorageFolder? createdFolder = null;
+			try
+			{
+				createdFolder = await rootFolder.CreateFolderAsync(folderName);
+				Assert.AreEqual(folderName, createdFolder.DisplayName);
+			}
+			finally
+			{
+				if (createdFolder != null)
+				{
+					await createdFolder.DeleteAsync();
+				}
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+		public async Task When_CreateFolder_With_File_Like_Name()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var folderName = GetRandomTextFileName();
+			StorageFolder? createdFolder = null;
+			try
+			{
+				createdFolder = await rootFolder.CreateFolderAsync(folderName);
+				Assert.AreEqual(folderName, createdFolder.Name);
+				Assert.AreEqual(folderName, createdFolder.DisplayName);
```

Comment:
> The user-friendly name of the current folder.

I wouldn't compare a user-friendly name to the actual name.

Reply:
In this case it is the same, so it is safe